zcash_client_sqlite/src/error.rs

```diff
@@ -147,7 +147,7 @@ impl fmt::Display for SqliteClientError {
             SqliteClientError::BlockConflict(h) => write!(f, "A block hash conflict occurred at height {}; rewind required.", u32::from(*h)),
             SqliteClientError::NonSequentialBlocks => write!(f, "`put_blocks` requires that the provided block range be sequential"),
             SqliteClientError::DiversifierIndexOutOfRange => write!(f, "The space of available diversifier indices is exhausted"),
-            SqliteClientError::AccountUnknown(acct_id) => write!(f, "Account {} does not belong to this wallet.", u32::from(*acct_id)),
+            SqliteClientError::AccountUnknown(acct_id) => write!(f, "Account {} does not belong to this wallet.", acct_id.0),
```

Comment:
This error message is now confusing because it exposes an internal detail of the wallet without clarification (the caller cannot do anything with this information, as they (correctly) cannot construct `zcash_client_sqlite::AccountId` themselves). The error should either contain a user-meaningful account identifier (such as could potentially be extracted from the current `Account` enum per my previous comments), or just not print the crate-internal account ID in the error message and rely on the caller to know which account this refers to by dead reckoning.